src/batch/src/executor/monitor/stats.rs

```diff
@@ -11,16 +14,140 @@
+                            delete_cache.push(collector);
+                        }
+                        Err(tokio::sync::mpsc::error::TryRecvError::Empty) => {
+                            break;
+                        }
+                        Err(_) => {
+                            // Error handle need modify later.
+                            error!("delete_queue_receiver is closed");
+                            connect = false;
+                            break;
+                        }
+                    }
+                }
+            }
+        });
+
+        Self {
+            registry,
+            sender: delete_queue_sender,
+        }
+    }
+
+    pub fn create_task_metrics(&self, id: TaskId) -> BatchTaskMetrics {
+        BatchTaskMetrics::new(self.registry.clone(), id, Some(self.sender.clone()))
+    }
+
+    /// Create a new `BatchTaskMetricsManager` instance used in tests or other places.
+    pub fn unused() -> Self {
+        let (delete_queue_sender, _) = tokio::sync::mpsc::channel::<Box<dyn Collector>>(1);
+        Self {
+            sender: delete_queue_sender,
+            registry: prometheus::Registry::new(),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct BatchTaskMetrics {
+    sender: Option<Sender<Box<dyn Collector>>>,
+    pub exchange_recv_row_number: GenericCounterVec<AtomicU64>,
+}
+
+impl BatchTaskMetrics {
+    pub fn new(registry: Registry, id: TaskId, sender: Option<Sender<Box<dyn Collector>>>) -> Self {
+        let opt = {
+            let const_labels = HashMap::from([
+                ("query_id".to_string(), id.query_id),
+                ("target_staget_id".to_string(), id.stage_id.to_string()),
+                ("target_task_id".to_string(), id.task_id.to_string()),
+            ]);
+            opts!(
+                "batch_exchange_recv_row_number",
+                "Total number of row that have been received from upstream source",
+            )
+            .const_labels(const_labels)
+        };
+        let exchange_recv_row_number = register_int_counter_vec_with_registry!(
+            opt,
+            &["source_stage_id", "source_task_id"],
+            registry
+        )
+        .unwrap();
+        Self {
+            sender,
+            exchange_recv_row_number,
+        }
+    }
+
+    /// This function execute after the exucution done.
+    /// Send all the record to the delete queue.
+    pub async fn clear_record(&self) {
+        if let Some(sender) = self.sender.as_ref() {
+            if sender
+                .send(Box::new(self.exchange_recv_row_number.clone()))
+                .await
+                .is_err()
+            {
+                error!("Failed to send delete record to delete queue");
+            }
+        }
+    }
+
+    pub fn unused() -> Self {
+        Self::new(prometheus::Registry::new(), TaskId::default(), None)
+    }
+}
 
 pub struct BatchMetrics {
     pub row_seq_scan_next_duration: Histogram,
```

Comment:
I think we should move it to executor, but as discussed offline, we can put it in another pr.